crates/testing-framework/src/services/results.rs

```diff
@@ -36,13 +36,16 @@ impl ResultStore {
 }
 
 impl MarineService for ResultStore {
-    fn call(&self, params: CallRequestParams) -> FunctionOutcome {
+    fn call(&self, mut params: CallRequestParams) -> FunctionOutcome {
         let results = self.results.borrow();
-        if let Some((_, suffix)) = params.service_id.split_once("..") {
-            if let Ok(key) = suffix.parse() {
-                let service_desc = results.get(&key).expect("Unknown result id");
-                let service_result = service_desc.call(params);
-                FunctionOutcome::ServiceResult(service_result, Duration::ZERO)
+        if let Some((real_service_id, suffix)) = params.service_id.rsplit_once("..") {
+            if let Ok(result_id) = suffix.parse() {
+                let service_desc = results.get(&result_id).unwrap_or_else(|| {
```

Comment:
let's decrease identation level here with `match`

Reply:
Like this?

Reply:
Yep, thanks